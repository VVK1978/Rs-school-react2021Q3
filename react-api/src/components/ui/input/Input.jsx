import React from 'react';

export default function Input() {
  const handlerChange = () => {};
  return (
    <input
      type="text"
      className="search-input"
      onChange={handlerChange}
      name="search"
      placeholder="Search by name..."
    />
  );
}
