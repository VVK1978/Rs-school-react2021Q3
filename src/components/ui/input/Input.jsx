import React from 'react';
import { useDispatch } from 'react-redux';
import { setSearchValue } from '../../reducers/characterReducer';

export default function Input() {
  const dispatch = useDispatch();

  const handlerChange = (event) => {
    dispatch(setSearchValue(event.target.value));
  };
  return (
    <>
      <input
        type="text"
        className="search-input"
        onChange={handlerChange}
				name="search"
				placeholder='Search by name...'
      />
    </>
  );
}
