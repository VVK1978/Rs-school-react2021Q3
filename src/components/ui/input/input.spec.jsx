/* eslint-disable no-undef */
import React from 'react';
import shallow from 'enzyme/build/shallow';
import * as redux from 'react-redux';
import Input from './Input.jsx';

describe('Component with redux hooks', () => {
  let spyOnUseSelector;
  let spyOnUseDispatch;
  let mockDispatch;

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector');
    spyOnUseSelector.mockReturnValue([{ id: 1, text: 'Old Item' }]);
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
    mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render Input component', () => {
    const component = shallow(<Input />);
    expect(component).toMatchSnapshot();
  });
});
