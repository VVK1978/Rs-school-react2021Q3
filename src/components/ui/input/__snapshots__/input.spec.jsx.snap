// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component with redux hooks should render Input component 1`] = `ShallowWrapper {}`;
